qiskit/providers/backend.py

```diff
@@ -229,3 +240,435 @@ class can handle either situation.
             Job: The job object for the run
         """
         pass
+
+
+class QubitProperties:
+    """A representation of the properties of a qubit on a backend."""
+
+    __slots = ("t1", "t2", "frequency", "properties")
+
+    def __init__(self, t1=None, t2=None, frequency=None, properties=None):
+        """Create a new ``QubitProperties`` object
+
+        Args:
+            t1: The T1 time for a qubit in seconds
+            t2: The T2 time for a qubit in seconds
+            frequency: The frequency of a qubit in Hz
+            properties: A free form dictionary of additional properties the
+                backend has for a qubit.
+        """
+        self.t1 = t1
+        self.t2 = t2
+        self.frequency = frequency
+        self.properties = properties
+
+    def __repr__(self):
+        return (
+            f"QubitProperties(t1={self.t1}, t2={self.t2}, "
+            f"frequency={self.frequency}, properties={self.properties})"
+        )
+
+
+class BackendV2(Backend, ABC):
+    """Abstract class for Backends
+
+    This abstract class is to be used for all Backend objects created by a
+    provider. This version differs from earlier abstract Backend classes in
+    that the configuration attribute no longer exists instead attributes
+    exposing equivalent required immutable properties of the backend device
+    are added. For example ``backend.configuration().n_qubits`` is accessible
+    from ``backend.num_qubits`` now.
+
+    The ``options`` attribute of the backend is used to contain the dynamic
+    user configurable options of the backend. It should be used more for
+    runtime options that configure how the backend is used. For example,
+    something like a ``shots`` field for a backend that runs experiments which
+    would contain an int for how many shots to execute.
+
+    If migrating a provider from :class:`~qiskit.providers.BackendV1` or
+    :class:`~qiskit.providers.BaseBackend` one thing to keep in mind is for
+    backwards compatibility you might need to add a configuration method that
+    will build a :class:`~qiskit.providers.models.BackendConfiguration` object
+    and :class:`~qiskit.providers.models.BackendProperties` from the attributes
+    defined in this class for backwards compatibility.
+    """
+
+    version = 2
+
+    def __init__(
+        self,
+        provider,
+        name: str = None,
+        description: str = None,
+        online_date: datetime.datetime = None,
+        backend_version: str = None,
+        **fields,
+    ):
+        """Initialize a BackendV2 based backend
+
+        Args:
+            provider (Provider): A backwards reference to the
```

Comment:
How can we make a device mock for simulator or unittest? Any fake provider object will be provided?

Reply:
Hmm, maybe I should just remove this, I don't think it's really needed. I left it in because `BaseBackend` and `BackendV1` both had it. For the mock backend I created I just set this to `None`: https://github.com/Qiskit/qiskit-terra/pull/5885/files#diff-36c9171b4f5709acd5034e14b32c1aa450d4c9168c9a9a4164129478d1986938R34

Reply:
Make it optional seems better than explicitly giving `None`.

Reply:
I made it optional in: https://github.com/Qiskit/qiskit-terra/pull/5885/commits/099a77b54c3ff7db2c62c69481bd92368c480cb6